### PR TITLE
HomeWork Week3

### PR DESCRIPTION
**Pull request by YaseirAlkhwalda**

### CODE REVIEW

File: Week3/homework/step2-6.js [hunk 1]
```diff
@@ -4,13 +4,19 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
...
 
 function printArray3d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  if (Array.isArray(arr)) {
+    arr.forEach(elem => printArray3d(elem));
+  } else {
+    console.log(arr);
+  }
 }
```
Comment (remarcmij) [inline Week3/homework/step2-6.js]: 👍 for using a recursive function (that we did not explain in class) that supports arrays of any dimension. You are the first student to do this. 

However, I do see some code duplication here (not DRY). Can you change this code such that you still have two functions `printArray2d` and `printArray3d` (required to pass the unit tests), but with the duplicated code factored out into a third function?

File: Week3/homework/step2-7.js [hunk 2]
```diff
@@ -21,3 +21,5 @@ f2(y);
 console.log(y);
 
 // Add your explanation as a comment here
+// In the first function the value of x does not change because x passed by Value.
+// while in the second function the value of y.x changes because y passed by Reference.
```
Comment (remarcmij) [inline Week3/homework/step2-7.js]: 👍 for your explanation

Comment (YaseirAlkhwalda) [reply Week3/homework/step2-7.js]: Thank you very much for your comments. I update my homework :").
Best Regards

**Comment by YaseirAlkhwalda** [comment]
> LGTM (Looks Good To Me)

Thanks :")

**Pull request closed without merging**